workflow/executor/executor_test.go

```diff
@@ -382,6 +382,13 @@ func TestSaveArtifacts(t *testing.T) {
 	we.Template.Outputs.Artifacts[0].Optional = false
 	err = we.SaveArtifacts(ctx)
 	assert.Error(t, err)
+
+	we.Template.Outputs.Artifacts[0].Optional = true
```

Comment:
Thanks for review, I'm not sure if I want to append a single test here, please give me some advice.

Reply:
> because `SaveArtifacts` actually deletes the artifact after saving, maybe you can wrap this whole test into a forloop, and use different archive strategy for each for loop element e.g. https://github.com/argoproj/argo-workflows/blob/master/persist/sqldb/archived_workflow_labels_test.go#L31

Thanks for your suggestion, I will resubmit later